lib/utils.js

```diff
@@ -13,6 +13,44 @@ module.exports = {
     });
   },
 
+  getApiGatewayStageName() {
+    let stageName = {};
+    let stageNameStage, stageNameDeploy, keyStage, keyDeploy, keyDeployFromStage;
+
+    const serviceTemplate = this.serverless.service.resources;
```

Comment:
do we need to check that there are resources, or that some plugin did not add Stages to compiledCloudFormationTemplate programatically?

Reply:
Here read `AWS::ApiGateway::Stage` from `this.serverless.service.resources` not `compiledCloudFormationTemplate` is because the plugin hooks with event `'before:aws:package:finalize:mergeCustomProviderResources'`: https://github.com/serverless/serverless/blob/c345e9a9bfb5dd49a13788009430e24c400243f2/lib/plugins/aws/package/lib/mergeCustomProviderResources.js#L17
if some other plugins add Stages to `compiledCloudFormationTemplate` before this event, the stage name will be the one in `this.serverless.service.resources`. If some plugin add Stages to `compiledCloudFormationTemplate` after this event, then it seems we could not get that stage name

Reply:
Makes sense, this is fine